### DISCUSSION TITLE
Software Functions...

### DISCUSSION THREAD

**Discussion by ic-racer** [srozum/film_camera_tester#48] [Ideas]
When analyzing focal plane shutters it would be nice if the knob could simply be rotated back and forth to see all the calculations on the 3 'results' screens without having to fire the shutter again or start/stop the mode.  

Also, on the 'average' function, couldn't it just keep a running average for how ever many times the shutter is fired without having to specify the number beforehand?

Combining the two ideas, the 'average' screen could go away and one could either be in "instant" or "average" mode when viewing the information on the three calculation screens. "average" mode just counting up and averaging all the results until it is reset.

**Comment by srozum** [comment]
Thanks for your suggestions. I thought about it too. I feel like I've explained it before, but current UI is dictated by available memory limit with chosen microprocessor. Implementing such behavior will require dynamic memory allocation, which is not safe with Arduino, and to sacrifice some test mode to free the memory. What I already done, for example, in v2.8 by removing "Efficiency interpolation for leaf shutters" in order to free memory for Exposure mode.

Yes, choosing different MCU would solve the problem but then it will be a different project. There are plenty of similar projects in the field, like Matt (aka Reveni Labs) and others, I hope they will do better.

**Reply by Surreal62** [reply]
@srozum "I hope they will do better."
You do yourself a disservice. They may do 'different', but I'd not say 'better', as your tester does everything and more than a professional camera repairer needs to accomplish testing. 
You are to be applauded! 👍👍👍👍

**Comment by ic-racer** [comment]
Yes, you did a good job to cram all the programming into the device. 
Now that I have built the device, I don't know how I got along without it! It is pretty incredible.

**Comment by ic-racer** [comment]
The Reveni looks promising, but the level of attention to making the focal plane sensors will be important. 
My Reveni meter had the light sensor misplaced. I hollowed out the opening to improve the offset view angle. 
![DSC_0004 2](https://github.com/srozum/film_camera_tester/assets/128929440/69659ee5-0e34-4026-a513-dbfbd18be733)
